smoke/test0.sh

```diff
@@ -186,6 +190,19 @@ ${CLI} verify -S 1 -f $MPT/test1 || fail "verify test1 after replay"
 ${CLI} verify -S 2 -f $MPT/test2 || fail "verify test2 after replay"
 ${CLI} verify -S 3 -f $MPT/test3 || fail "verify test3 after replay"
 
+# Check the behavior when system_uuid file is lost(renamed)
+${MV_SYS_UUID} $SYS_UUID_FILE $SYS_UUID_FILE"_old" || fail "mv of system-uuid to _old failed"
+${CLI} mkmeta $DEV				   || fail "mkmeta with no sys-uuid file should pass"
+${CLI} logplay $MPT 				   || fail "logplay with no sys-uuid file should pass"
+${CLI} creat -r -s 4096 $MPT/file1 		   && fail "creat with no sys-uuid file should fail"
+${CLI} mkdir $MPT/sys_uuid_dir			   && fail "mkdir with no sys-uuid file should fail"
+sudo dd if=/dev/random of=$TEST_F bs=1M count=4	   || fail "creation of test file should pass"
+${CLI} cp $TEST_F $MPT/copy			   && fail "cp with no sys-uuid file should fail"
+${CLI} clone $MPT/test1 $MPT/test1_clone	   && fail "clone with no sys-uuid file should fail"
+sudo rm $SYS_UUID_FILE				   || fail "delete of new sys-uuid file should pass"
+sudo rm $TEST_F					   || fail "delete of test file should pass"
+${MV_SYS_UUID} $SYS_UUID_FILE"_old" $SYS_UUID_FILE || fail "mv of _old to system-uuid failed"
+
```

Comment:
I don't object to smoke test coverage for this, but I think unit test is also essential. I think we could cover every failure mode without writing too much code there (by redirecting the location of the uuid file into /tmp/whatever, we can make it valid but not readable, invalid, invisible, etc.)

It's very important not to blow away the file if it's valid, even if we don't have access.

Reply:
Agreed, added unit tests also. 
But, famfs_create_sys_uuid_file() checks for the existence of the uuid file and if it is not there, creates the file and populates it with the uuid, Checking for invalid or unreadable uuid does not fall into it. We can simulate it by adding new tests.